tests/testthat/test-helpers.R

```diff
@@ -214,3 +225,77 @@ test_that("h_join_models throws the error for non-braced expression", {
     "Assertion on 'body\\(model1\\)' failed: Must inherit from class '\\{', but has class 'call'." # nolintr
   )
 })
+
+# h_format_number ----
+
+test_that("h_format_number works as expected", {
+  result <- c(
+    h_format_number(0.0001),
+    h_format_number(20000, digits = 3),
+    h_format_number(20000, prefix = "P", suffix = "S")
+  )
+  expected <- c("1.00000E-04", "2.000E+04", "P2.00000E+04S")
+
+  expect_identical(result, expected)
+})
+
+test_that("h_format_number works as expected when no change", {
+  result <- c(
+    h_format_number(1),
+    h_format_number(1, digits = 3),
+    h_format_number(1, prefix = "P", suffix = "S")
+  )
+  expected <- c(1, 1, 1)
+
+  expect_identical(result, expected)
+})
+
+# h_rapply ----
+
+test_that("h_rapply works as expected", {
+  my_model <- function() {
+    alpha0 <- mean(1:10)
+    alpha1 <- 600000
+  }
+  # Replace format of numbers using `formatC` function.
+  result <- h_rapply(
+    x = body(my_model),
+    fun = formatC,
+    classes = c("integer", "numeric"),
+    digits = 3,
+    format = "E"
+  )
+  expected_fun <- function() {
+    alpha0 <- mean("1.000E+00":"1.000E+01")
+    alpha1 <- "6.000E+05"
+  }
+  expected <- body(expected_fun)
+
+  expect_identical(result, expected)
+})
+
+# h_write_model ----
+
+test_that("h_write_model works as expected", {
+  my_model <- function() {
+    alpha0 <- mean(1:10)
+    alpha1 <- 600000
+  }
+
+  h_write_model(my_model, "./functions_output/h_write_model.jags", 5)
+  expect_snapshot(readLines("./functions_output/h_write_model.jags"))
```

Comment:
please use a temporary file instead (see `?tempfile`), we cannot just write to any directory during tests